### PR TITLE
Feature/db

### PR DESCRIPTION
**Pull request by radeklos**

### CODE REVIEW

**Comment by radeklos** [comment]
@peeklondon @krak3n CR please. I will fix the build later on there is an issue with building a container.

File: Dockerfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ ENV PATH $PATH:$GOPATH/bin
 
 # OS Dependencies
 RUN echo 'http://dl-4.alpinelinux.org/alpine/edge/main' >> /etc/apk/repositories \
-    && apk update && apk add go go-tools git ca-certificates make bash && rm -rf /var/cache/apk/*
+    && apk update && apk add go go-tools git gcc g++ ca-certificates make bash && rm -rf /var/cache/apk/*
```
Comment (krak3n) [inline Dockerfile]: `build-base` adds everything you should need.

File: api/handler.go [hunk 2]
```diff
@@ -3,24 +3,168 @@ package api
...
+	buf := new(bytes.Buffer)
+	buf.ReadFrom(r.Body)
+	rep, err := serializer.UnmarshalReport(buf.Bytes())
+	if err != nil {
+		log.Error("Unmarshal report: ", err)
+		http.Error(w, "oops", 422)
+		return
+	}
+
+	rrr := Report{
```
Comment (krak3n) [inline api/handler.go]: Rarrrrrrrrr? :stuck_out_tongue: 

Maybe a more descriptive variable name would be better here.

**Comment by krak3n** [comment]
The comments I posted are not required. Looks cool :+1:

**Pull request merged**